File: src/main/java/lana/thingService/thing/ThingService.java
package lana.thingService.thing;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface ThingService {

    Optional<Thing> update(Thing thing);

    Optional<Thing> find(Integer id);

    Thing create(Thing thing);

    void delete(Integer id);

    Page<Thing> findAll(Pageable pageable);
}

File: src/main/java/lana/thingService/thing/ThingService.kt
package lana.thingService.thing

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.domain.Specification

interface ThingService {

    fun update(thing: Thing): Thing?

    fun create(thing: Thing): Thing?

    fun find(id: Int): Thing?

    fun delete(id: Int)

    fun findAll(pageable: Pageable): Page<Thing>

    fun findAll(specification: Specification<Thing>, pageable: Pageable): Page<Thing>

}
